pkg/errors/request.go

```diff
@@ -27,11 +27,11 @@ import (
 )
 
 // ValidateUnaryContext validates that a context for a unary request is valid
-// and contains all required information.
+// and contains all required information, and returns a YARPC error with code
+// yarpcerrors.CodeInvalidArgument otherwise.
 func ValidateUnaryContext(ctx context.Context) error {
```

Comment:
I'm open to disagreeing and committing, but I want to explain my reasoning:

My reasoning being that error *constructors* should be in the errors package, but the ValidateUnaryContext function is not constructing an error, it's performing validation and depending on conditions will create an error, which can be said for just about every piece of code we have (which is why I'm against this as a precedent).

e.g. let's say I have a function that takes an input method name, and if the method name is improperly formatted it returns an error, otherwise it returns some split of the name.  Under the current logic, that function is creating an error and should be in the errors package, which is not necessarily what we want to do.

In this case, I would probably bundle the ExpectEncodings and ValidateUnaryContext under a "validate" package instead.

`validate.Encodings`
`validate.UnaryContext`